### PR TITLE
Update DataSourcePoller import since after refactor in core

### PR DESCRIPTION
**Pull request by ilyasfoo**
### Changes proposed in this Pull Request:

Related to https://github.com/woocommerce/woocommerce/issues/44451 and https://github.com/woocommerce/woocommerce/pull/45547

This PR updates import after a refactor in core.

### How to test the changes in this Pull Request:

1. Set up a Woo Express site with WoA

#### Test with current WooCommerce (8.7.0 as time of writing)
1. After uploading this bridge PR, ensure loading homescreen does not crash

#### Test with WooCommerce 8.8.0-rc:
3. Go to `_cli`, and then run the following:
    - `wp plugin delete woocommerce`
    - `wp plugin install https://github.com/woocommerce/woocommerce/releases/download/8.8.0-rc.1/woocommerce.zip`
4. Go to homescreen and ensure it does not crash
 

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by rjchow** [comment]
Drive-by reminder not to forget about this 😆

**Comment by ilyasfoo** [comment]
@rjchow Haha thanks! It seems like the release is going to be on 9th April, while I'm AFK. Perhaps an alternative would be to detect if the class exists and dynamically require the file 🤔

File: includes/notes/data/class-wc-calypso-bridge-admin-note-data-store.php [hunk 1]
```diff
@@ -35,6 +34,27 @@ protected function has_allow_list() {
...
+	public function fetch_specs() {
+		$possible_paths = [
+			'Automattic\WooCommerce\Admin\RemoteInboxNotifications\DataSourcePoller',
+			'Automattic\WooCommerce\Admin\RemoteInboxNotifications\RemoteInboxNotificationsDataSourcePoller',
+		];
+
+		foreach ( $possible_paths as $class_path ) {
+			if ( class_exists( $class_path ) ) {
+				// Directly call the static method from the fully qualified class name.
+				return $class_path::get_instance()->get_specs_from_data_source();
```
Comment (chihsuan) [inline includes/notes/data/class-wc-calypso-bridge-admin-note-data-store.php:51]:
```suggestion
				return $class_path::get_instance()->get_specs_from_data_sources();
```

File: includes/notes/data/class-wc-calypso-bridge-admin-note-data-store.php [hunk 2]
```diff
@@ -35,6 +34,27 @@ protected function has_allow_list() {
 		return wc_calypso_bridge_is_ecommerce_trial_plan();
```
Comment (chihsuan) [inline includes/notes/data/class-wc-calypso-bridge-admin-note-data-store.php:34]: I changed this to return `true` to test this on my Woo Express site.

**Review by chihsuan — CHANGES_REQUESTED** [review]
I got `PHP Fatal error:  Uncaught Error: Call to undefined method Automattic\WooCommerce\Admin\RemoteInboxNotifications\DataSourcePoller::get_specs_from_data_source()`

I think the we should call `get_specs_from_data_sources` method instead. 🙂

File: includes/notes/data/class-wc-calypso-bridge-admin-note-data-store.php (hunk 2 above)
Comment (ilyasfoo) [reply includes/notes/data/class-wc-calypso-bridge-admin-note-data-store.php:34]: Good call!

**Comment by ilyasfoo** [comment]
Thanks for catching all those mistakes, @chihsuan! I've applied all suggested changes, please take another look 🙏

File: includes/notes/data/class-wc-calypso-bridge-admin-note-data-store.php [hunk 3]
```diff
@@ -35,6 +34,27 @@ protected function has_allow_list() {
...
 
+	/**
+	 * Attempts to fetch specs from the first available DataSourcePoller class.
+	 *
+	 * @return array The specs data or null if not retrievable.
+	 */
+	public function fetch_specs() {
+		$possible_paths = [
+			'Automattic\WooCommerce\Admin\RemoteInboxNotifications\DataSourcePoller',
+			'Automattic\WooCommerce\Admin\RemoteInboxNotifications\RemoteInboxNotificationsDataSourcePoller',
```
[the remark this answers is not in the document]
Comment (ilyasfoo) [reply includes/notes/data/class-wc-calypso-bridge-admin-note-data-store.php]: Makes sense, thanks!

**Review by chihsuan — APPROVED** [review]
Thanks, tested well! 🚢

**Pull request merged**